lib/nerves_network/dhcp_manager.ex

```diff
@@ -2,49 +2,31 @@ defmodule Nerves.Network.DHCPManager do
   use GenServer
   require Logger
   import Nerves.Network.Utils
-  alias Nerves.Network.Types
 
   @moduledoc false
 
+  # The current state machine state is called "context" to avoid confusion between server
+  # state and state machine state.
   defstruct context: :removed,
             ifname: nil,
             settings: nil,
             dhcp_pid: nil,
             dhcp_retry_interval: 60_000,
             dhcp_retry_timer: nil
 
-  @typedoc "Settings for starting the server."
-  @type dhcp_settings :: Nerves.Network.setup_settings
-
-  @typep context :: Types.interface_context | :dhcp | :dhcp_retry
-
-  @typedoc """
-  The current state machine state is called "context" to avoid confusion between server
-  state and state machine state.
-  """
-  @type t :: %__MODULE__{
-    context: context,
-    ifname: Types.ifname | nil,
-    settings: dhcp_settings,
-    dhcp_pid: GenServer.server() | nil,
-    dhcp_retry_interval: integer,
-    dhcp_retry_timer: reference
-  }
-
-  @doc false
-  @spec start_link(Types.ifname, dhcp_settings, GenServer.options) :: GenServer.on_start
   def start_link(ifname, settings, opts \\ []) do
     Logger.debug fn -> "DHCPManager starting.... ifname: #{inspect ifname}; settings: #{inspect settings}" end
     GenServer.start_link(__MODULE__, {ifname, settings}, opts)
   end
 
   def init({ifname, settings}) do
-    # Register for nerves_network_interface and udhcpc events
+    # Register for nerves_network_interface and dhclient events
     {:ok, _} = Registry.register(Nerves.NetworkInterface, ifname, [])
-    {:ok, _} = Registry.register(Nerves.Udhcpc, ifname, [])
+    {:ok, _} = Registry.register(Nerves.Dhclientv4, ifname, [])
```

Comment:
To me it is good enough.. We may need some better way if we would like to upstream this change. Mike or Amos perhaps would know more how to do it fancily ;)